spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpSendingMessageHandler.java

```diff
@@ -226,11 +218,38 @@ public void setConnectionFactory(AbstractConnectionFactory connectionFactory) {
 	@Override
 	public void addNewConnection(TcpConnection connection) {
 		this.connections.put(connection.getConnectionId(), connection);
+
+		logger.debug(System.currentTimeMillis() + "," + LocalDateTime.now() + ",add," + connection.getConnectionId() + "," + this.connections.size() + "," +
+				this.hashCode());
+		int leakageCount = 0;
+		if (connections.size() > MAXIMUM_CONNECTIONS) {
```

Comment:
We will not be using this change; I will fix the `AbstractConnectionFactory` as discussed in https://github.com/spring-projects/spring-integration/issues/3701#issuecomment-1003135701

Reply:
OK